statusHandler/persister/persistentHandler.go

```diff
@@ -0,0 +1,200 @@
+package persister
+
+import (
+	"sync"
+	"time"
+
+	"github.com/ElrondNetwork/elrond-go/core"
+	"github.com/ElrondNetwork/elrond-go/core/check"
+	"github.com/ElrondNetwork/elrond-go/data/typeConverters"
+	"github.com/ElrondNetwork/elrond-go/logger"
+	"github.com/ElrondNetwork/elrond-go/marshal"
+	"github.com/ElrondNetwork/elrond-go/process"
+	"github.com/ElrondNetwork/elrond-go/storage"
+	"github.com/ElrondNetwork/elrond-go/storage/storageUnit"
+)
+
+var log = logger.GetOrCreate("statusHandler/persister")
+
+// PersistentStatusHandler is a status handler that will save metrics in storage
+type PersistentStatusHandler struct {
+	store                    storage.Storer
+	persistentMetrics        *sync.Map
+	marshalizer              marshal.Marshalizer
+	uint64ByteSliceConverter typeConverters.Uint64ByteSliceConverter
+	startSaveInStorage       bool
+}
+
+// NewPersistentStatusHandler will return an instance of the persistent status handler
+func NewPersistentStatusHandler(
+	marshalizer marshal.Marshalizer,
+	uint64ByteSliceConverter typeConverters.Uint64ByteSliceConverter,
+) (*PersistentStatusHandler, error) {
+	if check.IfNil(marshalizer) {
+		return nil, process.ErrNilMarshalizer
+	}
+	if check.IfNil(uint64ByteSliceConverter) {
+		return nil, process.ErrNilUint64Converter
+	}
+
+	psh := new(PersistentStatusHandler)
+	psh.store = storageUnit.NewNilStorer()
+	psh.uint64ByteSliceConverter = uint64ByteSliceConverter
+	psh.marshalizer = marshalizer
+	psh.persistentMetrics = &sync.Map{}
+	psh.initMap()
+
+	go func() {
```

Comment:
Why do you need this go routine and sleep inside? Why you didn't set psh.startSaveInStorage to true in one single line?

Reply:
Because when node start, nonce metric is chaging. And if modified will save in storage wrong data. So i have to wait 1 second when node starting, before i save metrics in storage.

Reply:
How are you sure that 1 second is enough?

Reply:
How are you sure that 1 second is enough?